zerver/tests/test_email_mirror.py

```diff
@@ -217,6 +218,68 @@ def test_filter_footer_many_parts(self) -> None:
         # Multiple possible footers, don't strip
         self.assertEqual(result, text)
 
+
+class TestStreamEmailTruncation(ZulipTestCase):
+    def test_stream_email_message_with_topic_length_under_max_length(self) -> None:
+        user_profile = self.example_user('hamlet')
+        self.login(user_profile.email)
+        self.subscribe(user_profile, "Denmark")
+        stream = get_stream("Denmark", user_profile.realm)
+
+        stream_to_address = encode_email_address(stream)
+
+        incoming_valid_message = MIMEText('TestStreamEmailWithTrucnatedTopic Body')
+
+        incoming_valid_message['Subject'] = 'TestSubject<MAX_LENGTH'
+        incoming_valid_message['From'] = self.example_email('hamlet')
+        incoming_valid_message['To'] = stream_to_address
+        incoming_valid_message['Reply-to'] = self.example_email('othello')
+
+        process_message(incoming_valid_message)
+        message = most_recent_message(user_profile)
+        self.assertEqual(message.topic_name(), incoming_valid_message['Subject'])
+        self.assertEqual(message.content, "TestStreamEmailWithTrucnatedTopic Body")
+
+    def test_stream_email_message_with_topic_length_equal_max_length(self) -> None:
+        user_profile = self.example_user('hamlet')
+        self.login(user_profile.email)
+        self.subscribe(user_profile, "Denmark")
+        stream = get_stream("Denmark", user_profile.realm)
+
+        stream_to_address = encode_email_address(stream)
+
+        incoming_valid_message = MIMEText('TestStreamEmailWithTrucnatedTopic Body')
+
+        incoming_valid_message['Subject'] = '**************TestSubject=MAX_LENGTH************endMAXLENGTH'
```

Comment:
I think it'd be clearer to set the subject here and in the other tests below as ``'X' * SOME_LENGTH``, as that makes it easier to know the length